e2e/testconfig/testconfig.go

```diff
@@ -223,3 +219,43 @@ func defaultModifyGenesis() func(ibc.ChainConfig, []byte) ([]byte, error) {
 		return bz, nil
 	}
 }
+
+// modifyGovAppState takes the existing gov app state and marshals it to either a govv1 GenesisState
+// or a govv1beta1 GenesisState depending on the simapp version.
+func modifyGovAppState(chainConfig ibc.ChainConfig, govAppState json.RawMessage) ([]byte, error) {
+	cfg := testutil.MakeTestEncodingConfig()
+
+	cdc := codec.NewProtoCodec(cfg.InterfaceRegistry)
+	govv1.RegisterInterfaces(cfg.InterfaceRegistry)
+	govv1beta1.RegisterInterfaces(cfg.InterfaceRegistry)
```

Comment:
I wonder if we should have a types directory? `testsuite.Codec()` is another function that we could use here but would run into import cycle issues

As a side note, `testsuite.codecAndEncodingConfig` should also start by using the [`DefaultEncoding`](https://github.com/strangelove-ventures/ibctest/pull/337/files#diff-97215974790dcffb1bd4e12b6e7e73d0fb968d91a22eb5fd157371803a46737cR29) function

Reply:
Good point about the `DefaultEncoding`. I wonder should we open a pr to add variable args to default encoding to allow for `additionalEncodings`, that seems like it would be the most flexible option. WDYT?